presto-common/src/main/java/com/facebook/presto/common/type/LongEnumType.java

```diff
@@ -114,25 +123,22 @@ public boolean equals(Object o)
 
             LongEnumMap other = (LongEnumMap) o;
 
-            return Objects.equals(this.enumMap, other.enumMap);
+            return Objects.equals(typeName, other.typeName) && Objects.equals(enumMap, other.enumMap);
         }
 
         @Override
         public String toString()
         {
-            return "enum:bigint{"
-                    + enumMap.entrySet()
-                    .stream()
+            return format("%s{%s}", typeName, enumMap.entrySet().stream()
```

Comment:
Is the order the same as the user given? That might be good.

Reply:
Depending on how it's stored at the source it might or might not be possible to have the order as "user given". Since we are implementing it as map, that's not really possible. I don't want to make this like a map block that's too complicated. 😂